apis/v1alpha1/generator.yaml

```diff
@@ -161,3 +161,47 @@ resources:
       Tags:
         compare:
           is_ignored: true
+  User:
+    hooks:
+      delta_pre_compare:
+        code: compareTags(delta, a, b)
+      sdk_read_one_post_set_output:
+        template_path: hooks/user/sdk_read_one_post_set_output.go.tpl
+      sdk_create_post_set_output:
+        template_path: hooks/user/sdk_create_post_set_output.go.tpl
+      sdk_update_post_set_output:
+        template_path: hooks/user/sdk_update_post_set_output.go.tpl
+      sdk_delete_pre_build_request:
+        template_path: hooks/user/sdk_delete_pre_build_request.go.tpl
+    exceptions:
+      terminal_codes:
+        - InvalidInput
+    renames:
+      operations:
+        CreateUser:
+          input_fields:
+            UserName: Name
+        GetUser:
+          input_fields:
+            UserName: Name
+        UpdateUser:
+          input_fields:
+            UserName: Name
```

Comment:
I think this may need to be reexamined due to the weirdness of how IAM implements user renaming upstream.  The update API endpoint wants "UserName" to be the current/existing value and "NewUserName" to be the new value.

c.f. https://github.com/aws-controllers-k8s/iam-controller/pull/44/files#r1010960864